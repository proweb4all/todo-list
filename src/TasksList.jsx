import React from 'react';
import PropTypes from 'prop-types';
import Task from './Task';

const TasksList = ({ tasks, onChangeCheck, onDeleteTask }) => {
  const sortedList = tasks.slice().sort((a, b) => a.done - b.done)
  return (
    <ul className='list' tasks={tasks}>
      {sortedList.map(task => (
        <Task
          key={task.id}
          {...task}
          onChange={onChangeCheck}
          onDelete={onDeleteTask}
        />
      ))}
    </ul>
  );
}

TasksList.propTypes = {
  tasks: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      done: PropTypes.bool,
      id: PropTypes.string
    }),
  ).isRequired,
  onChangeCheck: PropTypes.func.isRequired,
  onDeleteTask: PropTypes.func.isRequired
}

export default TasksList;
