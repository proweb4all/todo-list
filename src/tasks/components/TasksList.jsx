import React from 'react';
import PropTypes from 'prop-types';
import Task from './Task';

const TasksList = ({ tasks, onChangeCheck, onDeleteTask }) => {
  return (
    <ul className='list' tasks={tasks}>
      {tasks.map(task => (
        <Task
          key={task.id}
          {...task}
          onChange={onChangeCheck}
          onDelete={onDeleteTask}
        />
      ))}
    </ul>
  );
}

TasksList.propTypes = {
  tasks: PropTypes.arrayOf(
    PropTypes.shape({
      text: PropTypes.string,
      done: PropTypes.bool,
      id: PropTypes.string
    }),
  ).isRequired,
  onChangeCheck: PropTypes.func.isRequired,
  onDeleteTask: PropTypes.func.isRequired
}

export default TasksList;
