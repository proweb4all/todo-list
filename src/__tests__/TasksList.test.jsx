import React from 'react'
import TasksList from '../TasksList'
import {shallow} from 'enzyme'
import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
Enzyme.configure({ adapter: new Adapter() });

describe('<TasksList />', () => {
  it('should display todolist', () => {
    const props = {
      onChangeCheck: jest.fn(), 
      onDeleteTask: jest.fn(),
      tasks: [
        {text: 'Task 1', done: true, id: 'task-id-1'},
        {text: 'Task 2', done: false, id: 'task-id-2'},
        {text: 'Task 3', done: true, id: 'task-id-3'},
      ], 
    }
    const wrappedComponent = shallow(<TasksList {...props} />)
    expect(wrappedComponent).toMatchSnapshot()
  })
})