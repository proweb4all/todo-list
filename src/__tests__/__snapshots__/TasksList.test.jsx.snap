// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<TasksList /> should display todolist 1`] = `ShallowWrapper {}`;
